### ISSUE TITLE
BC Break | All migrations are hardly bound to MySQL

### ISSUE THREAD

**Issue by stloyd** [Sylius/Sylius#13188]
**Sylius version affected**: every?

**Description**  
All migrations that are shipped in Sylius are not usable for other database than MySQL ones (MariaDB works as it's same family), so you cannot use them on i.e. Postgresql, yet all those migrations are marked as "not limited" to MySQL

**Steps to reproduce**  
For example recent one about catalog promotions:
https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/Migrations/Version20211001073918.php#L23
```php
$this->addSql('CREATE TABLE sylius_catalog_promotion_scope (id INT AUTO_INCREMENT NOT NULL, promotion_id INT DEFAULT NULL, type VARCHAR(255) NOT NULL, configuration LONGTEXT NOT NULL COMMENT \'(DC2Type:array)\', INDEX IDX_584AA86A139DF194 (promotion_id), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB');
```

Here is how it looks manually generated on Postgre:
```php
$this->addSql('DROP SEQUENCE sylius_catalog_promotion_rule_id_seq CASCADE');
$this->addSql('CREATE SEQUENCE sylius_catalog_promotion_scope_id_seq INCREMENT BY 1 MINVALUE 1 START 1');
$this->addSql('CREATE TABLE sylius_catalog_promotion_scope (id INT NOT NULL, promotion_id INT DEFAULT NULL, type VARCHAR(255) NOT NULL, configuration TEXT NOT NULL, PRIMARY KEY(id))');
$this->addSql('CREATE INDEX IDX_584AA86A139DF194 ON sylius_catalog_promotion_scope (promotion_id)');
$this->addSql('COMMENT ON COLUMN sylius_catalog_promotion_scope.configuration IS \'(DC2Type:array)\'');
$this->addSql('ALTER TABLE sylius_catalog_promotion_scope ADD CONSTRAINT FK_584AA86A139DF194 FOREIGN KEY (promotion_id) REFERENCES sylius_catalog_promotion (id) NOT DEFERRABLE INITIALLY IMMEDIATE');
```

**Possible Solution**  
Either mark migrations as platform specific & put clear disclaimer that those should be not used on other dbs than MySQL alike.

Or create approach to generate those migrations for both or more database types.

**Comment by stloyd** [comment]
@lchrusciel This was fixed by #13379 & #13380, thanks!

**Comment by lchrusciel** [comment]
Hopefully, we can support PostgreSQL better in a near future as well :)
